Documentation/gitignore.txt

```diff
@@ -89,23 +89,26 @@ PATTERN FORMAT
    Put a backslash ("`\`") in front of the first "`!`" for patterns
    that begin with a literal "`!`", for example, "`\!important!.txt`".
 
- - If the pattern ends with a slash, it is removed for the
-   purpose of the following description, but it would only find
-   a match with a directory.  In other words, `foo/` will match a
-   directory `foo` and paths underneath it, but will not match a
+ - If the pattern ends with a slash, it would only find
+   a match with a directory. In other words, `foo/` will match a
+   directory `foo` anywhere, but will not match a
    regular file or a symbolic link `foo` (this is consistent
    with the way how pathspec works in general in Git).
 
- - If the pattern does not contain a slash '/', Git treats it as
-   a shell glob pattern and checks for a match against the
-   pathname relative to the location of the `.gitignore` file
-   (relative to the toplevel of the work tree if not from a
-   `.gitignore` file).
+ - If the pattern contains a slash '/' other then a trailing slash,
+   it is always considered from the root. In other words, `foo/bar`
```

Comment:
I also suspect that "root" maybe a confused location. In Linux/*nix, root is a special place, while on Windows (Git-for-) root either doesn't exist (MS view), or could be anywhere (each bash has its own) depending on view point.

I think it actually means the location of the .gitignore file (and we can have many, repeated at many levels).

The whole .gitignore doc has a long history of confusion...

Reply:
This is true. Would you think its better to write "[...] is always considered from the location of the `.gitignore` file"?

Reply:
'It' is certainly worth suggesting. Though maybe "_the pattern_ is always considered from the `.gitignore` file location".

Also note that Git PRs, by them selves, are ignored. Once the checks are OK, one can send them upstream by either send-email, or via gitgitgadget.

I believe the documentation failure maybe that there has bee an update of the toolset. see https://public-inbox.org/git/20190329123520.27549-6-szeder.dev@gmail.com/

Reply:
Thank you for your help. Does it mean I have to create a fork at https://github.com/gitgitgadget/gitgitgadget, and create a PR with the same content and this PR? And about the documentation error, does it mean I simply have to wait till it gets fixed?